README.md

```diff
@@ -4,46 +4,205 @@
+1 - it adds a unique request ID to each request (if none is present in the specified header);
+2 - outputs the log in a semi-structured format that can easily be parsed by log aggregation tools;
+
+In your ktor module definition (e.g: [mobi.waterdog.rest.template.tests.Application](https://github.com/waterdog-oss/ktor-template/blob/development/ktor-template-example/src/main/kotlin/mobi/waterdog/rest/template/tests/Application.kt))
+In your ktor module definition (e.g: [mobi.waterdog.rest.template.tests.Application](https://github.com/waterdog-oss/ktor-template/blob/development/ktor-template-example/src/main/kotlin/mobi/waterdog/rest/template/tests/Application.kt))
+```kotlin
+val callIdHeader = SemiStructuredLogFormatter.REQUEST_ID_HEADER
+install(CallLogging) {
+    level = org.slf4j.event.Level.INFO
+    callIdMdc(callIdHeader)
+}
+install(CallId) {
+    generate { it.request.headers[callIdHeader] ?: UUID.randomUUID().toString() }
+    replyToHeader(callIdHeader)
+}
+```
 
-## Preparing a production build
+You can then use your usual slf4j logging, and the output of the logs should be something like:
+```
+2020-11-05T23:50:42.665Z INFO  de1126db-5314-49bd-ba2b-24c1e07d6f8e 200 OK: GET - /readiness                                 {"timestamp":"2020-11-05T23:50:42.665Z","level":"INFO","thread":"DefaultDispatcher-worker-3 @request#1","logger":"ktor.test","message":"200 OK: GET - /readiness","metadata":{"X-Request-Id":"de1126db-5314-49bd-ba2b-24c1e07d6f8e"}}
+```
 
-In order to build the docker image you can simply run `docker build -t ktor-template .`. To run the application just use
-the docker run command: `docker run -p 8080:8080 -it ktor-template:latest`.
+#### Readiness and liveness checks:
+The health check feature uses concepts from: [https://github.com/zensum/ktor-health-check]
 
-If you cloned this repo and changed the project name and version you may need to pass specific build args if you haven't 
-overridden the defaults on the dockerfile. For additional information regarding docker build arguments check the `--build-arg` 
-flag on the official [docker build documentation](https://docs.docker.com/engine/reference/commandline/build/).
+In your ktor module definition (e.g: [mobi.waterdog.rest.template.tests.Application](https://github.com/waterdog-oss/ktor-template/blob/development/ktor-template-example/src/main/kotlin/mobi/waterdog/rest/template/tests/Application.kt))
+```kotlin
+install(Health) {
+    liveness()
+    readiness()
+}
+```
 
-When running the image there are a number of environment variables that can be and should be used in production.
+The `liveness` and `readiness` functions are just plain kotlin extension functions, and you can find an example [here](https://github.com/waterdog-oss/ktor-template/blob/development/ktor-template-tests/src/main/kotlin/mobi/waterdog/rest/template/tests/core/utils/healthcheck/HealthCheck.kt)
```

Comment:
This will change from `ktor-template-tests` to `ktor-template-examples` right?

Reply:
yup, obviously :)